src/s1reader/s1_burst_slc.py

```diff
@@ -360,6 +389,68 @@ class represents a polynomial function of range
 
         return az_carrier_poly
 
+    def total_doppler(self, xstep=500, ystep=50):
+        """
+        Compute total Doppler which is the sum of two components: 
+        (1) the geometrical Doppler induced by the relative movement 
+        of the sensor and target
+        (2) the TOPS specicifc Doppler caused by the electric steering 
+        of the beam along the azimuth direction resulting in Doppler varying
+        with azimuth time. 
+        Parameters
+        ----------
+        xstep: int
+            Spacing along x direction [pixels]
+        ystep: int
+            Spacing along y direction [pixels]
+
+        Returns
+        -------
+        2D array: float
+           Total Doppler which is the sum of the geometrical Doppler and 
+           beam steering induced Doppler [Hz]
+        """
+        x = np.arange(0, self.width, xstep, dtype=int)
+        y = np.arange(0, self.length, ystep, dtype=int)
+        x_mesh, y_mesh = np.meshgrid(x, y)
+        kt, eta, eta_ref = compute_az_carrier_frequency(self, self.orbit,
+                                        offset=0.0,
+                                        position=(y_mesh, x_mesh))
+
+        antenna_steering_Doppler = kt*(eta - eta_ref)
+
+        slant_range = self.starting_range + x * self.range_pixel_spacing
+        geometrical_doppler = self.doppler.poly1d.eval(slant_range)
+
+        total_Doppler = antenna_steering_Doppler + geometrical_doppler
+
+        return x, y, total_Doppler
+
+    def range_delay_caused_by_doppler_shift(self, xstep=500, ystep=50):
```

Comment:
may be `doppler_shift_range_delay`?

Reply:
I was not very proud of the name too. But is "doppler_shift_range_delay" clear on "range delay caused by Doppler shift" and not confused by "Doppler shift caused by range delay"? 
I will change to your suggestion for now but let me know if you or yunjun find a better name!

Reply:
how about `doppler_induced_range_shift()`, which is basically the subsection title in the ETAD [ATBD page 44](https://sentinels.copernicus.eu/documents/247904/4629150/Sentinel-1-ETAD-Algorithm-Technical-Baseline-Document.pdf) ?

Reply:
@yunjunz I like that :D